File: benchmarks/suite/julia.py
import time

t0 = time.perf_counter()

c_re = -0.7
c_im = 0.27015
total = 0
for py in range(2000):
    zi_init = -1.5 + py * 3.0 / 2000.0
    for px in range(2000):
        zr = -1.5 + px * 3.0 / 2000.0
        zi = zi_init
        iter = 0
        while iter < 50:
            if zr * zr + zi * zi > 4.0:
                break
            new_zr = zr * zr - zi * zi + c_re
            zi = 2.0 * zr * zi + c_im
            zr = new_zr
            iter += 1
        total += iter

t1 = time.perf_counter()
print(total)
print(f"elapsed: {t1 - t0:.3f}s")

File: benchmarks/suite/julia.c
#include <stdio.h>
#include <time.h>

int main(void) {
    struct timespec t0, t1;
    clock_gettime(CLOCK_MONOTONIC, &t0);

    double c_re = -0.7;
    double c_im = 0.27015;
    long long total = 0;
    for (int py = 0; py < 2000; py++) {
        double zi_init = -1.5 + py * 3.0 / 2000.0;
        for (int px = 0; px < 2000; px++) {
            double zr = -1.5 + px * 3.0 / 2000.0;
            double zi = zi_init;
            int iter = 0;
            while (iter < 50) {
                if (zr * zr + zi * zi > 4.0) break;
                double new_zr = zr * zr - zi * zi + c_re;
                zi = 2.0 * zr * zi + c_im;
                zr = new_zr;
                iter++;
            }
            total += iter;
        }
    }

    clock_gettime(CLOCK_MONOTONIC, &t1);
    double elapsed = (t1.tv_sec - t0.tv_sec) + (t1.tv_nsec - t0.tv_nsec) / 1e9;
    printf("%lld\n", total);
    printf("elapsed: %.3fs\n", elapsed);
    return 0;
}
